Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeMask.azsl

```diff
@@ -35,45 +56,60 @@ struct VSOutput
     // Base fields (required by the template azsli file)...
     // "centroid" is needed for SV_Depth to compile
     precise linear centroid float4 m_position : SV_Position;
+
+    // Fragment depth
+    float m_depth : UV0;
 };
 
 VSOutput MainVS(VSInput IN)
 {
-    VSOutput OUT;    
-    float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0)).xyz;
-    OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(worldPosition, 1.0));
-
-    // Offset the depth of redrawn entities to avoid z fighting with the underlying entity rendered data
-    // A proper programatic solution will be investigated in LYN-10304
-    OUT.m_position.z += 0.001;
-
-/*
-    For use in LYN-9929
+    VSOutput OUT;
 
-    // NDCs
-    float3 ndc = float3(OUT.m_position.x / OUT.m_position.w, OUT.m_position. y/ OUT.m_position.w, OUT.m_position.z / OUT.m_position.w);
-
-    // Normalized screenspace UVs
-    float2 uv = float2((ndc.x + 1.0) * 0.5, (ndc.y + 1.0) * 0.5);
+    const float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0)).xyz;
+    OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(worldPosition, 1.0));
+    
+    // The existing depth buffer contents are the result of the resolved MSAA depth buffer so contain depth contributions
+    // from surrounding sub-fragments so we need to pull the current fragment towards the viewer slightly in order to reduce
+    // z fighting with the sub-fragment depth contributions
+    OUT.m_position.z *= 1.01;
 
     // Fragment depth
-    float depth = ndc.z;
-*/
-
+    OUT.m_depth = OUT.m_position.z / OUT.m_position.w;
     return OUT;
 }
 
 struct PixelOutput
 {
-    float4 m_color : SV_Target0;
+    float4 m_mask : SV_Target0;
```

Comment:
Format is R8G8_UNORM but you still have float4 here, you need to change it to match the texture specified in the pass file

Reply:
Good catch! Fixed 👍